src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,337 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to and navigating through them.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub mod project_name;
+
+pub use breadcrumb::Breadcrumb;
+pub use project_name::ProjectName;
+
+use enso_frp as frp;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use ensogl::display::shape::text::text_field::FocusManager;
+use std::cmp;
+use enso_protocol::language_server::MethodPointer;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+// FIXME[dg] hardcoded literal for glyph of height 12.0. Copied from port.rs
+const GLYPH_WIDTH       : f32 = 7.224_609_4;
+const VERTICAL_MARGIN   : f32 = GLYPH_WIDTH;
+const HORIZONTAL_MARGIN : f32 = GLYPH_WIDTH;
+const TEXT_SIZE         : f32 = 12.0;
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Push breadcrumb.
+    pub push_breadcrumb : frp::Source<(Rc<MethodPointer>,uuid::Uuid)>,
+    /// Pop breadcrumb.
+    pub pop_breadcrumb : frp::Source,
+}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            def push_breadcrumb = source();
+            def pop_breadcrumb  = source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb}
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    pub breadcrumb_push : frp::Source<(Rc<MethodPointer>,uuid::Uuid)>,
```

Comment:
ditto

Reply:
And I think We should use here some named structure, analogous to the `ide::model::execution_context::LocalCall` (or even somehow reused here). I don't see what is uuid required by the API.